### PR TITLE
Adding validation to check not to have ddl, dml commands in sqlSearch filter text

### PR DESCRIPTION
**Pull request by nazeer1100126**

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SQLInjectionValidator {
+
+	private final static String[] DDL_COMMANDS = { "create", "drop", "alter", "truncate", "comment" };
+
+	private final static String[] DML_COMMANDS = { "select", "insert", "update", "delete", "merge", "upsert", "call" };
+
+	private final static String[] COMMENTS = { "--", "({", "/*" };
```
Comment (alexivanov) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java:31]: **#** could also be used for commenting out queries

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+	private final static String SQL_PATTERN = "[a-zA-Z_, ()0-9]*";
```
Comment (alexivanov) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java:33]: **=** , **'** / **"** should be matched. **%** could also be added to the list.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		while (inputSqlString.indexOf("= ") > 0) { //Don't remove space after = operator
+			inputSqlString = inputSqlString.replaceAll("= ", "=");
+		}
+
+		StringTokenizer tokenizer = new StringTokenizer(inputSqlString, " ");
+		while (tokenizer.hasMoreTokens()) {
+			String token = tokenizer.nextToken().trim();
+			if (token.equals("'")) {
```
Comment (alexivanov) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java:69]: Spaces before and after **"** / **'** would be nice to be added as they can be in the middle or prepended/appended to a token. e.g `column = "foo"`

**Review by alexivanov — COMMENTED** [review]
Overall a good approach :+1:

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java (hunk 2 above)
Comment (nazeer1100126) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java:33]: @MegaAlex are you taking about escape char \ or / ?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java (hunk 3 above)
Comment (nazeer1100126) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java:69]: @MegaAlex I am removing the spaces before and after of '='. I believe above case won't appear.

**Pull request closed without merging**